### PR TITLE
Adding prices to spools

### PR DESCRIPTION
**Pull request by pdsccode**
As requested in https://github.com/Donkie/Spoolman/issues/88, prices for single spools are practical for instances where a new spool should be added that has been bought for a different price.

### CODE REVIEW

**Comment by Donkie** [comment]
Thank you for your contribution!

I think it would be nice if the displayed price in the UI would default to the filament's price if the spool doesn't have any price set.
You should be able to mutate the price field in the `collapseSpool` function in list.tsx to achieve this.

Also please add integration tests for this in tests_integration/tests/spool

**Comment by pdsccode** [comment]
Good idea with the default. Didn't think about this. Will add it and also the tests.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 default_language_version:
-  python: python3.9
+  python: python3
```
Comment (Donkie) [inline .pre-commit-config.yaml:2]: Undo this change

File: client/public/locales/en/common.json [hunk 2]
```diff
@@ -150,6 +151,7 @@
             "archived": "Archived"
         },
         "fields_help": {
+            "price": "Price of a full spool in the system configured currency.",
```
Comment (Donkie) [inline client/public/locales/en/common.json:154]: Here you should add that if this is not set, the filament's price will be assumed instead.

File: client/src/pages/spools/show.tsx [hunk 3]
```diff
@@ -58,6 +58,8 @@ export const SpoolShow: React.FC<IResourceComponentsProps> = () => {
       <NumberField value={record?.id ?? ""} />
       <Title level={5}>{t("spool.fields.filament")}</Title>
       <TextField value={record ? filamentURL(record?.filament) : ""} />
+      <Title level={5}>{t("spool.fields.price")}</Title>
+      <NumberField value={record?.price ?? ""} />
```
Comment (Donkie) [inline client/src/pages/spools/show.tsx:62]: Here you should also default to the filament's price, to make it consistent with the list view

**Review by Donkie — CHANGES_REQUESTED** [review]

**Comment by Donkie** [comment]
Tested it locally now and it generally seems to work fine. I have some minor comments I want you to address before I merge it :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (pdsccode) [reply .pre-commit-config.yaml]: I changed it back now. Accidentally worked with 3.11 and got pre-commit not running without this change. Works now.

File: client/public/locales/en/common.json (hunk 2 above)
Comment (pdsccode) [reply client/public/locales/en/common.json]: Added as requested.

File: client/src/pages/spools/show.tsx (hunk 3 above)
Comment (pdsccode) [reply client/src/pages/spools/show.tsx]: Filament price is now shown on spools/show.tsx when spool price is not set :)

**Comment by Donkie** [comment]
Sorry for the delay, it's merged now. Thank you for the help! :)

**Comment by pdsccode** [comment]
No worries. Thanks for letting me help and advise me as well. 😊
